### PR TITLE
 Add month calendar in SideNav

### PR DESCRIPTION
**Pull request by lkudyba**
@quirk0o @mhib @jtkpiotr @adamzima @s1mplex

### CODE REVIEW

File: client/app/components/calendar/layout/SideCalendar.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                onChange={this.updateSelectedDate} />
+    );
+  }
+
+  _setMondayAsFirstWeekDay() {
+    moment.updateLocale('en', {
+      week: {
+        dow: 1
+      }
+    });
```
Comment (quirk0o) [inline client/app/components/calendar/layout/SideCalendar.jsx]: This doesn't seem like a good place to put global locale initialization.
Maybe put it in `Register.jsx` ? Or separate config file that you load in `Register.jsx`?

File: client/app/components/calendar/layout/SideCalendar.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+export default class SideCalendar extends React.Component {
+  static contextTypes = {
+    router: React.PropTypes.object
+  };
+
+  constructor(...args) {
+    super(...args);
+    this._setMondayAsFirstWeekDay();
+    this.state = { value: moment() };
```
Comment (quirk0o) [inline client/app/components/calendar/layout/SideCalendar.jsx]: You should not store date here - there should be once source of truth about the date being currently displayed in the calendar - the date from the router query.
Inject it into `SideCalendar` as a prop

File: client/test/components/calendar/layout/SideCalendar.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import moment from 'moment';
+
+describe('<SideCalendar />', () => {
+  const spy = sinon.spy();
+  const routerContext = { router: { push: spy } };
+  const wrapper = shallow(<SideCalendar />, { context: routerContext });
+  const stringDate = '2015-01-01';
+  const date = moment(stringDate);
+
+  describe('_routeToSelectedDate()', () => {
```
Comment (quirk0o) [inline client/test/components/calendar/layout/SideCalendar.spec.jsx:16]: Don't test private component methods T_T

**Review by quirk0o — CHANGES_REQUESTED** [review]

File: client/app/components/calendar/layout/SideCalendar.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  onSelect(date) {
+    this._routeToSelectedDate(date);
+    this.updateSelectedDate(date);
+  }
+
+  updateSelectedDate(date) {
+    this.setState({ value: moment(date) });
+  }
+
+  disableWeekend(date) {
```
Comment (madejejej) [inline client/app/components/calendar/layout/SideCalendar.jsx]: This method doesn't disable weekends. It returns true if the date is a weekday. I'd call it `isWeekday`

**Review by madejejej — APPROVED** [review]

File: client/app/components/calendar/layout/SideCalendar.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.state = { value: moment(this.props.date) };
+
+    bindAll(this, ['onSelect', 'updateSelectedDate']);
+  }
+
+  onSelect(date) {
```
Comment (quirk0o) [inline client/app/components/calendar/layout/SideCalendar.jsx:27]: Convention is to start handler names with `handle..` ;)

File: client/app/components/calendar/table/Calendar.jsx [hunk 6]
```diff
@@ -66,9 +67,13 @@ export default class Calendar extends React.Component {
     );
   }
 
+  selectedDay(day) {
```
Comment (quirk0o) [inline client/app/components/calendar/table/Calendar.jsx]: Maybe name is `isSelectedDay`?

**Review by quirk0o — APPROVED** [review]

File: client/test/components/calendar/layout/SideCalendar.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+describe('<SideCalendar />', () => {
+  const spy = sinon.spy();
+  const routerContext = { router: { push: spy } };
+  const initialDate = moment();
+  const wrapper = shallow(<SideCalendar date={initialDate} />, { context: routerContext });
+  const stringDate = '2015-01-01';
+  const date = moment(stringDate);
+
+  describe('_routeToSelectedDate()', () => {
```
Comment (quirk0o) [inline client/test/components/calendar/layout/SideCalendar.spec.jsx:16]: ;(

**Pull request merged**